### PR TITLE
Make block expiry summaries accurate

### PR DESCRIPTION
**Pull request by Muzzarino**

### CODE REVIEW

File: cmds/wiki/user.js [hunk 1]
```diff
@@ -321,19 +321,22 @@ function gamepedia_user(lang, msg, namespace, username, wiki, querystring, fragm
 			var isBlocked = false;
 			var blockedtimestamp = ( queryuser.blockedtimestamp ? dateformat.format(new Date(queryuser.blockedtimestamp)) : 'Invalid Date' );
 			var blockexpiry = queryuser.blockexpiry;
+			var isPermabanned = queryuser.isPermabanned;
```
Comment (Markus-Rost) [inline cmds/wiki/user.js:324]: `queryuser.isPermabanned` is always `undefined`, why is it used here?

File: i18n/en.json [hunk 2]
```diff
@@ -685,7 +685,8 @@
             "header": "$1 is currently blocked!",
             "noreason": "Blocked on $1 until $2 by $3.",
             "text": "Blocked on $1 until $2 by $3 with reason \"$4\".",
-            "until_infinity": "the end of all days"
+            "permabannednoreason": "Permanently blocked on $1 by $3."
+            "permabannedtext": "Permanently blocked on $1 by $3 with reason \"$4\".",
```
Comment (Markus-Rost) [inline i18n/en.json:689]: Invalid JSON and not sorted correctly (alphabetical by key).

`until_infinity` is still in use for IP blocks.

File: cmds/wiki/user.js [hunk 3]
```diff
@@ -321,19 +321,22 @@ function gamepedia_user(lang, msg, namespace, username, wiki, querystring, fragm
...
+				isPermabanned = false;
 				blockexpiry = dateformat.format(new Date(blockexpirydate));
 				if ( Date.parse(blockexpirydate) > Date.now() ) isBlocked = true;
 			}
 			var blockedby = queryuser.blockedby;
 			var blockreason = queryuser.blockreason;
 			var block = {
 				header: lang.get('user.block.header', username, queryuser.gender).escapeFormatting(),
-				text: lang.get('user.block.' + ( blockreason ? 'text' : 'noreason' ), blockedtimestamp, blockexpiry),
+				text: lang.get('user.block.' + ( blockreason ? ( isPermabanned ? 'permabannedtext' : 'text') : ( isPermabanned ? 'permabannednoreason' : 'noreason') ), blockedtimestamp, blockexpiry),
```
Comment (Markus-Rost) [inline cmds/wiki/user.js:339]:
```suggestion
				text: lang.get('user.block.' + ( isPermabanned ? 'permabanned' : '' ) + ( blockreason ? 'text' : 'noreason' ), blockedtimestamp, blockexpiry),
```

**Review by Markus-Rost — CHANGES_REQUESTED** [review]
MediaWiki is referring to those blocks as "indefinite" because it won't expire but could be manually unblocked at some point in the future. 

Using "permanent" here would be an own creation just like the current "until the end of all days".

Also, IP blocks have not been updated.

File: i18n/en.json (hunk 2 above)
Comment (Muzzarino) [reply i18n/en.json]: It's being replaced to make it correctly indicate a type of block if it's temp or not, but I've added a missing comma.

File: cmds/wiki/user.js (hunk 1 above)
Comment (Muzzarino) [reply cmds/wiki/user.js:324]: That's meant to be a boolean to indicate if the user's permanently banned which then uses an alternative message.

Comment (Markus-Rost) [reply cmds/wiki/user.js:324]: But `queryuser` has no `isPermabanned` property so it shouldn't be used here. It's literally the same as setting it to `undefined` but in a very ugly way.

Comment (Muzzarino) [reply cmds/wiki/user.js]: Changed it into a bool, how does it sound?

**Comment by Markus-Rost** [comment]
This has been added together with better duration display in b163039

**Pull request closed without merging**